src/content/insertable-streams/webgpu/js/multi_video_main.js

```diff
@@ -0,0 +1,271 @@
+
+'use strict';
+
+const wgslShaders = {
+    vertex: `
+struct VertexInput {
+  [[location(0)]] position : vec3<f32>;
+  [[location(1)]] uv : vec2<f32>;
+};
+
+struct VertexOutput {
+  [[builtin(position)]] Position : vec4<f32>;
+  [[location(0)]] fragUV : vec2<f32>;
+};
+
+[[stage(vertex)]]
+fn main(input : VertexInput) -> VertexOutput {
+    var output : VertexOutput;
+    output.Position = vec4<f32>(input.position, 1.0);
+    output.fragUV = vec2<f32>(-0.5,-0.0) + input.uv;
+    return output;
+}
+`,
+
+    fragment: `
+[[binding(0), group(0)]] var mySampler: sampler;
+[[binding(1), group(0)]] var myTexture: texture_2d<f32>;
+
+[[stage(fragment)]]
+fn main([[location(0)]] fragUV : vec2<f32>) -> [[location(0)]] vec4<f32> {
+  return textureSample(myTexture, mySampler, fragUV);
+}
+`,
+};
+
+async function updateScreenImage(source) {
+    let frame;
+    if (source) {
+        let { value: chunk } = await source.read();
+        frame = chunk;
+    }
+    return frame;
+}
+
+class WebGPUTransform {
+    constructor() {
+        this.canvas_ = null;
+        this.context_ = null;
+        this.device_ = null;
+        this.renderPipeline_ = null;
+        this.sampler_ = null;
+        this.videoTexture_ = null;
+        this.vertexBuffer_ = null;
+    }
+
+    async init() {
+        console.log('[WebGPUTransform] Initializing WebGPU.');
+
+        if (!this.canvas_) {
+            this.canvas_ = document.createElement('canvas');
+            document.getElementById('outputVideo').append(this.canvas_);
+            this.canvas_.width = 5000;
```

Comment:
src/css/main.css has a rule `canvas { max-width: 100%; }`, which means explicitly setting width and height here only sets the aspect ratio, so the actual rendered canvas has dimensions of 960x480, and squashed videos. How about just setting it to 960x540 here, so we get the right aspect ratios?